xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/migration/R140401000XWIKI15460DataMigration.java

```diff
@@ -174,7 +172,13 @@ private void internalHibernateMigrate(boolean isMainWiki) throws DataMigrationEx
                 .map(Map.Entry::getKey)
                 .collect(Collectors.toSet());
             for (String userReference : unknownUsers) {
-                this.store.deleteFilterPreferences(this.resolver.resolve(userReference));
+                DocumentReference userDocumentReference = this.resolver.resolve(userReference);
+                // Verify if the user still has some filters preferences to remove, in case they have already all been
+                // cleaned up when removing the filters preferences for the unknown wikis. Without this check, calling 
```

Comment:
I honestly don't see the point of complexity the code so much for this HSQLDB specific nonsense.